services-core/src/solution_submission.rs

```diff
@@ -201,26 +245,96 @@ fn extract_transaction_receipt(err: &MethodError) -> Option<&TransactionReceipt>
     }
 }
 
-fn convert_cancel_result(
-    result: Result<(), ExecutionError>,
-) -> Result<(), SolutionSubmissionError> {
-    let error = match result {
+fn convert_cancel_result(result: CancellationResult) -> Result<(), SolutionSubmissionError> {
+    let error = match result.0 {
         Ok(_) => anyhow!("solution submission transaction not confirmed in time"),
         Err(err) => Error::from(err).context("failed to cancel solution submission"),
     };
     Err(SolutionSubmissionError::Unexpected(error))
 }
 
-/*
+fn is_transaction_error(error: &ExecutionError) -> bool {
+    // This is the error as we've seen it on openethereum nodes. The code and error messages can
+    // be found in openethereum's source code in `rpc/src/v1/helpers/errors.rs`.
+    // TODO: check how this looks on geth and infura. Not recognizing the error is not a serious
+    // problem but it will make us sometimes log an error when there actually was no problem.
+    matches!(error, ExecutionError::Web3(Web3Error::Rpc(RpcError { code, .. })) if code.code() == -32010)
+}
+
+struct SolutionResult(Result<(), MethodError>);
+impl TransactionResult for SolutionResult {
```

Comment:
Maybe  `TransactionResult ` and `TransactionSending` could be implemented on `ethcontract::TransactionResult` and `ethcontract::TransactionBuilder` directly?

Reply:
Agree. Like I mentioned in last PR we can explore having an ethcontract and web3 feature in the retry standalone crate for more utility things.